dags/ala/cluster_setup.py

```diff
@@ -116,81 +170,100 @@ class PreIngestionEMRConfig(EMRConfig):
     Configurations: list = field(default_factory=lambda: [], init=False)
-    BootstrapActions: list = field(default_factory=lambda: [{
-        "Name": "Bootstrap action",
-        "ScriptBootstrapAction": {
-            "Args": [f"{ala_config.S3_BUCKET}"],
-            "Path": f"s3://{ala_config.S3_BUCKET}/airflow/dags/bootstrap-preingestion-actions.sh",
-        }
-    }], init=False)
+    BootstrapActions: list = field(
+        default_factory=lambda: [
+            {
+                "Name": "Bootstrap action",
+                "ScriptBootstrapAction": {
+                    "Args": [f"{ala_config.S3_BUCKET}"],
+                    "Path": f"s3://{ala_config.S3_BUCKET}/airflow/dags/bootstrap-preingestion-actions.sh",
+                },
+            }
+        ],
+        init=False,
+    )
+
+    def __post_init__(self):
+        super().__post_init__()
+        self.Tags += [{"Key": "product", "Value": "preingestion"}]
 
 
 @dataclass
 class PipelinesEMRConfig(EMRConfig):
-    """Base configuration for EMR cluster for pipelines. """
+    """Base configuration for EMR cluster for pipelines."""
+
     ReleaseLabel: str = field(default=ala_config.EMR_RELEASE, init=False)
-    Applications: list = field(default_factory=lambda: [
-        {'Name': 'Spark'},
-        {'Name': 'Hadoop'}], init=False)
-    Configurations: list = field(default_factory=lambda: [
-        {
-            "Classification": "spark",
-            "Properties": ala_config.SPARK_PROPERTIES}], init=False)
+    Applications: list = field(
+        default_factory=lambda: [{"Name": "Spark"}, {"Name": "Hadoop"}], init=False
+    )
+    Configurations: list = field(
+        default_factory=lambda: [
+            {"Classification": "spark", "Properties": ala_config.SPARK_PROPERTIES}
+        ],
+        init=False,
+    )
+
+    def __post_init__(self):
+        super().__post_init__()
+        self.Tags = self.Tags + [{"Key": "product", "Value": "pipeline"}]
 
 
 @dataclass
 class PipelinesSingleEMRConfig(PipelinesEMRConfig):
-    """Configuration for EMR cluster for pipelines with a single node. """
+    """Configuration for EMR cluster for pipelines with a single node."""
+
     instance_type: str
     ebs_size_in_gb: int
 
     @property
     def instance_groups(self):
         return [
             {
-                'Name': "Master nodes",
-                'Market': ala_config.MASTER_MARKET,
-                'InstanceRole': 'MASTER',
-                'InstanceType': self.instance_type,
-                'CustomAmiId': ala_config.PREINGESTION_AMI,
-                'InstanceCount': 1,
-                'EbsConfiguration': {
-                    'EbsBlockDeviceConfigs': [
+                "Name": "Master nodes",
+                "Market": ala_config.MASTER_MARKET,
+                "InstanceRole": "MASTER",
+                "InstanceType": self.instance_type,
+                "CustomAmiId": ala_config.PREINGESTION_AMI,
```

Comment:
I'm questioning how this would have come about after merging with branch 2_6. This was fixed in that branch. Is there anything else we miss from branch 2_6?